src/utils/s3-helpers.tsx

```diff
@@ -0,0 +1,70 @@
+import * as AWS from "aws-sdk";
+
+const S3_BUCKET = "models-resources";
+const S3_DIR_PREFIX = "glossary-resources";
+const S3_REGION = "us-east-1";
+const CLOUDFRONT_URL = "https://models-resources.concord.org";
+const CLOUDFRONT_DISTRIBUTION = "E1QHTGVGYD1DWZ";
+
+interface IParams {
+  dir: string;
+  filename: string;
+  accessKey: string;
+  secretKey: string;
+  body: AWS.S3.Types.Body;
+  contentType?: string;
+}
+
+export function s3Upload({ dir, filename, accessKey, secretKey, body, contentType = "" }: IParams) {
+  return new Promise((resolve, reject) => {
+    const s3 = new AWS.S3({
+      region: S3_REGION,
+      accessKeyId: accessKey,
+      secretAccessKey: secretKey,
+    });
+    const key = `${S3_DIR_PREFIX}/${dir}/${filename}`;
+    s3.upload({
```

Comment:
I think you could simplify this a little by using AWS's built in promise support. Here is the most official documentation I can found about it: https://aws.amazon.com/blogs/developer/support-for-promises-in-the-sdk/

Basically you can add `.promise()` to most every AWS api call.  Which means you can do something like:
`await s3.upload(...).promise();`

Reply:
Thanks, that definitely looks cleaner and won't be much work.